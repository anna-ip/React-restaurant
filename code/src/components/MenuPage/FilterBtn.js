import React, { useContext } from 'react'
import { MenuContext } from '../../App'
import styles from './FilterBtn.module.scss'

const FilterBtn = ({ filterMenuClick }) => {
  const [storeMenus] = useContext(MenuContext)
  const menus = storeMenus.menus

  const filteredBtn = ['all']
  menus.filter((item) => {
    const i = filteredBtn.findIndex((x) => x.category === item.category)
    if (i <= -1) {
      filteredBtn.push(item)
    }
    return null
  })

  return (
    <>
      {filteredBtn.map((section, index) => (
        <button
          onClick={(e) => filterMenuClick(e)}
          value={section.category || 'all'}
          key={index}
          className={styles.btn}
        >
          {section.category || 'all'}
        </button>
      ))}
    </>
  )
}

export default FilterBtn
