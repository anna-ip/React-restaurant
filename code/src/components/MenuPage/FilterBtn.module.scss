@import '../../styles/_variables.scss';

.btn {
  border: none;
  background: inherit;
  color: #c85935;
  text-transform: capitalize;

  @media (min-width: $layout-breakpoint-small) {
    font-size: 1.1rem;
  }
}