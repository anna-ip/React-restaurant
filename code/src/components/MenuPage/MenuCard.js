import React from 'react'
import styles from './MenuCard.module.scss'

const MenuCard = ({ filteredMenu }) => {
  return (
    <>
      {filteredMenu &&
        filteredMenu.map((menu) => (
          <div key={menu.id} className={styles.container}>
            <img
              src={`assets/${menu.img}`}
              alt="dish"
              width="150vw"
              height="150vw"
              className={styles.img}
            />
            <section className={styles.section}>
            <div className={styles.titleContainer}>
            <p className={styles.title}>{menu.title}</p>
            <p>{menu.price}</p>
            </div>
            <p className={styles.desc}>{menu.desc}</p>
            </section>
          </div>
        ))}
    </>
  )
}

export default MenuCard
