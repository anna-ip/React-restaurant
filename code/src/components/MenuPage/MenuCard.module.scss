@import '../../styles/_variables.scss';

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #e2ede9;
  max-width: 500px;
  padding: 0.5rem;
  margin: 0.5rem;

  @media (min-width: $layout-breakpoint-small) {
    flex-direction: row;
  }
}
.section {
  display: flex;
  @media (min-width: $layout-breakpoint-small) {
  flex-direction: column;
  margin: 0 0.5rem;
  width: 80%;
  }
}
.title {
  text-transform: capitalize;
}
.img {
  width: 20rem;
  height: auto;
  object-fit: cover;
  margin-top: 0.5rem;

  @media (min-width: $layout-breakpoint-small) {
    width: 10rem;
    height: 12rem;
  }
}
.titleContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 20rem;

   @media (min-width: $layout-breakpoint-small) {
     width: 80%;
   }
}
.desc {
  width: 20rem;
  text-align: initial;

}
.desc::first-letter {
  text-transform: capitalize;
}
