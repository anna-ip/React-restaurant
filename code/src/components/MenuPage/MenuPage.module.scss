@import '../../styles/_variables.scss';

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.link {
  text-decoration: none;
  font-size: 1.2rem;
}
.link:hover{
  text-decoration: underline;
}
.btnContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  width: 100%;
  border-top: 1px solid #c85935;
  border-bottom: 1px solid #c85935;
  padding: 0.5rem 0;
}
.cardContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 1.5rem 0;

  @media (min-width: $layout-breakpoint-small) {
    flex-direction: row;
    flex-wrap: wrap;
  }
}


