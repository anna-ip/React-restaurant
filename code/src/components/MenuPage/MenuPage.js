import React, { useContext, useState } from 'react'
import { Link } from 'react-router-dom'
import { MenuContext } from '../../App'
import FilterBtn from './FilterBtn'
import MenuCard from './MenuCard'
import styles from './MenuPage.module.scss'

const MenuPage = () => {
  const [storeMenus] = useContext(MenuContext)
  const menus = storeMenus.menus

  const [filteredMenu, setFilteredMenu] = useState(menus || [])

  const filterMenuClick = (e) => {
    const btnValue = e.target.value
    const filteredMenus = menus.filter((cat) => cat.category === btnValue)

    if (btnValue === 'all') {
      setFilteredMenu(menus)
    } else {
      setFilteredMenu(filteredMenus)
    }
  }

  return (
    <div className={styles.container}>
      <Link to="/">Go to home page</Link>

      <h2>Menu</h2>
      <div className={styles.btnContainer}>
        <FilterBtn filterMenuClick={filterMenuClick} />
      </div>
      <section className={styles.cardContainer}>
        <MenuCard filteredMenu={filteredMenu} />
      </section>
    </div>
  )
}

export default MenuPage
