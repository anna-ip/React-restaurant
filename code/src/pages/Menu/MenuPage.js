import React from "react";
import { MenuContext } from "../../App";
import FilterBtn from "../../components/MenuPageComponents/FilterBtn/FilterBtn";
import MenuCard from "../../components/MenuPageComponents/MenuCard/MenuCard";
import styles from "./MenuPage.module.scss";

const MenuPage = () => {
  const [storeMenus] = React.useContext(MenuContext);
  const menus = storeMenus.menus;
  const [filteredMenu, setFilteredMenu] = React.useState(menus || []);

  const filterMenuClick = (e) => {
    const btnValue = e.target.value;

    if (btnValue === "all") {
      setFilteredMenu(menus);
    } else {
      const filteredMenuItems = filterMenus(menus, btnValue);
      setFilteredMenu(filteredMenuItems);
    }
  };

  return (
    <div className={styles.container}>
      <div className={styles.btnContainer}>
        <FilterBtn filterMenuClick={filterMenuClick} />
      </div>
      <section className={styles.cardContainer}>
        <MenuCard filteredMenu={filteredMenu} />
      </section>
    </div>
  );
};

export const filterMenus = (menus, val) => {
  return menus.filter((menu) => menu.category === val);
};

export default MenuPage;
