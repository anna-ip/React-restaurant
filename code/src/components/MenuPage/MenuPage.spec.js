import React from 'react'
import renderer from 'react-test-renderer'
import MenuPage from './MenuPage'

test('<MenuPage/> renders correctly', () => {
  // const tree = renderer.create(<MenuPage menus={{ menus: '' }} />).toJSON()
  // expect(tree).toMatchSnapshot()
})
