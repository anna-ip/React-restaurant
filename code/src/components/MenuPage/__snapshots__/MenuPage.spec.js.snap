// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuPage/> Render Correctly 1`] = `
<div
  className="container"
>
  <Links
    className="link"
    text="Go to home page"
    to="/"
  />
  <h2>
    Menu
  </h2>
  <div
    className="btnContainer"
  >
    <FilterBtn
      filterMenuClick={[Function]}
    />
  </div>
  <section
    className="cardContainer"
  >
    <MenuCard
      filteredMenu={
        Array [
          Object {
            "category": "breakfast",
            "desc": "I'm baby woke mlkshk wolf bitters live-edge blue bottle, hammock freegan copper mug whatever cold-pressed ",
            "id": 1,
            "img": "pexels-ash-376464.jpg",
            "price": "250kr",
            "title": "buttermilk pancakes Gurka",
          },
          Object {
            "category": "lunch",
            "desc": "I'm baby woke mlkshk wolf bitters live-edge blue bottle, hammock freegan copper mug whatever cold-pressed ",
            "id": 2,
            "img": "pexels-ash-376464.jpg",
            "price": "250kr",
            "title": "biff rydberg",
          },
        ]
      }
    />
  </section>
</div>
`;
